src/models/tests/m_identity_tests.erl

```diff
@@ -0,0 +1,107 @@
+%% @author Maas-Maarten Zeeman <mmzeeman@xs4all.nl>
+%% @hidden
+
+-module(m_identity_tests).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("zotonic.hrl").
+
+hash_test() ->
+    ?assertMatch({bcrypt, <<"$2a$12$", _/binary>>}, m_identity:hash("1234")),
+    ?assertMatch({bcrypt, <<"$2a$12$", _/binary>>}, m_identity:hash(<<"1234">>)),
+    ?assertMatch({bcrypt, <<"$2a$12$", _/binary>>}, m_identity:hash(<<195, 154, 105, 105>>)),
+    ok.
+
+needs_rehash_test() ->
+    ?assertEqual(true, m_identity:needs_rehash(old_hash("password"))),
+    ?assertEqual(false, m_identity:needs_rehash(m_identity:hash("password"))),
+    ok.
+
+hash_is_equal_test() ->
+    PasswordHash = m_identity:hash("password"),
+
+    ?assert(m_identity:hash_is_equal("password", PasswordHash)),
+    ?assert(not m_identity:hash_is_equal("1234", PasswordHash)),
+
+    ok.
+
+hash_is_equal_old_hash_test() ->
+    PasswordHash = old_hash("password"),
+
+    % Validating stored old password hashes should still work.
+    ?assert(m_identity:hash_is_equal("password", PasswordHash)),
+    ?assert(not m_identity:hash_is_equal("1234", PasswordHash)),
+
+    ok.
+
+check_password_no_user() ->
```

Comment:
This is not run as it doesn’t have the `_test` function extension. Can you change that?